base/linalg/qr.jl

```diff
@@ -231,14 +242,20 @@ Stacktrace:
 [...]
 ```
 """
-qrfact!(A::StridedMatrix, ::Val{false}) = qrfactUnblocked!(A)
-qrfact!(A::StridedMatrix, ::Val{true}) = qrfactPivotedUnblocked!(A)
-qrfact!(A::StridedMatrix) = qrfact!(A, Val(false))
+function qrfact!(A::StridedMatrix, pivot = :none)
+    if pivot == :none
+        return qrfactUnblocked!(A)
+    elseif pivot == :colnorm
+        return qrfactPivotedUnblocked!(A)
```

Comment:
Transformations like these from explicit dispatch to dispatch-simulation-via-conditional seem a bit unfortunate. Perhaps some best-of-both-worlds approach exists, or could exist in the future? :)

Reply:
Perhaps dispatching to internal `Val` versions make the code cleaner (like https://github.com/JuliaLang/julia/pull/24960#issuecomment-349921490).

Not sure if there are any drawbacks.

Reply:
Nice! Thanks :). (Interestingly, that approach dovetails nicely with [Andy's comment](https://github.com/JuliaLang/julia/pull/25303#issuecomment-354538156).)

Reply:
I think the explicit list approach should be preferred Julian style for these methods

Reply:
Out of curiosity, why? :)